### PR TITLE
fix: [CDS-83115]: fix setup docs for agent with proxy configured

### PR DESCRIPTION
**Pull request by ashinsabu3**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/gitops/use-gitops/install-a-harness-git-ops-agent.md [hunk 1]
```diff
@@ -268,16 +268,28 @@ For steps on setting up the mapping and import, go to [Map Argo projects to Harn
 
 ## Proxy support
 
-The Harness GitOps Agent can work on environments where traffic is routed through a proxy. Perform the following steps to configure proxy support for the agent.
+The Harness GitOps Agent can work on environments where traffic is routed through a proxy. 
 
-1. Make sure that the agent is running in HTTP mode.  
+To enable proxy support for the Harness GitOps Agent in environments where traffic is routed through a proxy, configuration is required for two key components: the agent itself and the repo-server. Follow these steps to set up proxy support for both components.
```
Comment (michaelcretzman) [inline docs/continuous-delivery/gitops/use-gitops/install-a-harness-git-ops-agent.md:273]:
```suggestion
To enable proxy support for the Harness GitOps Agent in environments where traffic is routed through a proxy, configuration is required for two key components: the Agent and the Repository server. Follow these steps to set up proxy support for both components.
```

**Review by michaelcretzman — DISMISSED** [review]

File: docs/continuous-delivery/gitops/use-gitops/install-a-harness-git-ops-agent.md [hunk 2]
```diff
@@ -268,16 +268,28 @@ For steps on setting up the mapping and import, go to [Map Argo projects to Harn
...
+4. **ArgoCD Repo Server:** Add the following environment variables and relevant proxy details, such as URL, port, and auth details in the `argocd-repo-server` deployment. 
+  ```
+   - name: HTTPS_PROXY
+    value: "http://squid.proxy-test:3128"
+  - name: HTTP_PROXY
+    value: "http://squid.proxy-test:3128"
+  - name: NO_PROXY
+    value: "localhost,argocd-redis,127.0.0.1,$(KUBERNETES_SERVICE_HOST),({agentname}-agent)"
+  ``````
+
```
Comment (shivagowda) [inline docs/continuous-delivery/gitops/use-gitops/install-a-harness-git-ops-agent.md:292]: @ashinsabu3 is this tested? 
we have 2 init containers in repo-server, did they work with these proxy settings?

Comment (ashinsabu3) [reply docs/continuous-delivery/gitops/use-gitops/install-a-harness-git-ops-agent.md:292]: I'm adding the details for the other two containers as well.

Comment (shivagowda) [reply docs/continuous-delivery/gitops/use-gitops/install-a-harness-git-ops-agent.md:336]: image
there are 2 init containers.

**Review by shivagowda — APPROVED** [review]
lgtm

**Pull request merged**